doc/source/dsintro.rst

```diff
@@ -81,9 +81,26 @@ index is passed, one will be created having values ``[0, ..., len(data) - 1]``.
 
 **From dict**
 
-If ``data`` is a dict, if **index** is passed the values in data corresponding
-to the labels in the index will be pulled out. Otherwise, an index will be
-constructed from the sorted keys of the dict, if possible.
+.. note::
```

Comment:
Can you move this note a bit below? It seems strange to start this "from dict" section with a note without first explaining what it actually is. 
So I would give an example first, and only afterwards noting that the output depends on the python version